api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/GaugeHandler.java

```diff
@@ -149,7 +149,8 @@ public void getMetrics(
 
         Observable<Metric<Double>> metricObservable = null;
         if (tags != null) {
-            metricObservable = metricsService.findMetricsWithFilters(getTenant(), GAUGE, tags);
+            metricObservable = metricsService.findMetricIdentifiersWithFilters(getTenant(), GAUGE, tags)
+                    .flatMap(metricsService::findMetric);
```

Comment:
All of these ```findMetric``` calls are against the same partition. We ought to consider collecting all of the ids and doing a single query.